sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -133,7 +139,14 @@
   // We could consider making this a parameter.
   private static final int SPILLED_RECORD_SHARDING_FACTOR = 10;
 
+  // The record count and buffering duration to trigger flushing records to a tmp file. Mainly used
+  // for writing unbounded data to avoid generating too many small files.
+  private static final int FILE_TRIGGERING_RECORD_COUNT = 10000;
+  private static final Duration FILE_TRIGGERING_RECORD_BUFFERING_DURATION =
+      Duration.standardSeconds(1);
```

Comment:
These values are rather random :\ Let me know if you have insights of what values would be more proper.